File: model/src/main/java/cm/aptoide/pt/model/v7/base/BaseV7Response.java
package cm.aptoide.pt.model.v7.base;

import java.util.List;

/**
 * Created by neuro on 20-04-2016.
 */
public class BaseV7Response {

  private Info info;
  private List<Error> errors;

  public Error getError() {
    if (errors != null && errors.size() > 0) {
      return errors.get(0);
    } else {
      return null;
    }
  }

  public Info getInfo() {
    return info;
  }

  public void setInfo(Info info) {
    this.info = info;
  }

  public List<Error> getErrors() {
    return errors;
  }

  public void setErrors(List<Error> errors) {
    this.errors = errors;
  }

  public boolean isOk() {
    return info != null && info.getStatus() == Info.Status.OK;
  }
}

File: feature_search/src/main/java/cm/aptoide/pt/feature_search/data/network/baseresponse/BaseV7Response.kt
package cm.aptoide.pt.feature_apps.data.network.model

open class BaseV7Response {
  var info: Info? = null
  var errors: List<Error>? = null
  var error: Error? = null
    get() = if (errors != null && errors!!.isNotEmpty()) {
      errors!![0]
    } else {
      null
    }
  var isOk: Boolean = false
    get() = info != null && info!!.status == Info.Status.OK

  class Info(var status: Status?, var time: Time?) {

    enum class Status {
      OK, QUEUED, FAIL, Processing
    }

    class Time {
      var seconds = 0.0
      var human: String? = null
    }
  }

  class Error {
    var code: String? = null
    var description: String? = null
  }
}